### PR TITLE
docs(project): update to reflect v10 end of support

### LINKED ISSUE #12497: Update documentation to reflect the deprecation of `carbon-components` and `carbon-components-react`
As part of [the plan to deprecate these packages](https://github.com/carbon-design-system/carbon/issues/12494), the following documentation needs to be updated:

```[tasklist]
### Docs to update
- [x] Write a deprecation message to be used on npm and in the console, link to longform explanations
- [x] Create longform documentation explanations for both packages - see below
- [x] update both package readme's with a red github annotation
- [x] update guides/migration/v11.md
- [ ] ~update the files copyright header to include a deprecation notice?~
- [x] update anywhere else we reference these packages within the monorepo .md files
- [x] Update existing usages on carbon-website
- [x] Update banner on carbon-website current and v10
- [ ] ~platform/next (if applicable?)~
- [x] remove v10 links from the current storybook welcome page, replace with links to LTS, security policy, and other related links
- [x] update the v10 storybook banner to state it's deprecated
```

### Options for longform documentation explanations

Options include 
1. Create carbondesignsystem.com/deprecated-carbon-components and carbondesignsystem.com/deprecated-carbon-components-react that is not visible in the left nav
2. A GitHub discussion(s) (but maybe we don't want or need a comments section? or could the discussion prevent issues from being opened asking about the deprecation?)
3. A section on the package readme in addition to the deprecation notice at the top of the file

Benefits of 1 and 2, the GitHub options, include that we can get a permalink to a discussion/readme. Whereas on the website it might be easy to forget the pages exists and/or keep it updated.

### PR DESCRIPTION
**Pull request by tay1orjones**
Closes #12497
Related https://github.com/carbon-design-system/carbon-website/pull/4301

#### Changelog

**Changed**

- Update documentation to reflect v10 end of support for `carbon-components`, `carbon-components-react`, elements packages @ v10.x, etc.
- A few additional random updates of content that was outdated and/or incorrect.
- New links on the carbon react storybook welcome page

#### Testing / Reviewing

- Double check content, spelling, code, etc.

### CODE REVIEW

**Review by sstrubberg — APPROVED** [review]

File: .github/CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -89,9 +89,10 @@ members of the project's leadership.
     with the same questions can benefit.
 4.  If you have a pressing bug fix, it is best to make a PR directly to get your
     issues addressed.
-5.  `#carbon-components`: This channel is for questions about Carbon Components
-    only. It is not the place to ask general coding questions. Instead, use a
-    dev community like [StackOverflow](https://stackoverflow.com/).
+5.  `#carbon-react`, `#carbon-web-components`: This channel is for questions
```
Comment (alisonjoseph) [inline .github/CODE_OF_CONDUCT.md]:
```suggestion
5.  `#carbon-react`, `#carbon-web-components`: These channels are for questions
```

File: .github/CODE_OF_CONDUCT.md [hunk 2]
```diff
@@ -89,9 +89,10 @@ members of the project's leadership.
...
+    about Carbon Components only. It is not the place to ask general coding
```
Comment (alisonjoseph) [inline .github/CODE_OF_CONDUCT.md]:
```suggestion
    about Carbon Components only. They are not the place to ask general coding
```

**Review by alisonjoseph — APPROVED** [review]
LGTM! just a super tiny grammar nitpick. Nice update to the Storybook welcome links 🔥

File: .github/CODE_OF_CONDUCT.md (hunk 1 above)
Comment (tay1orjones) [reply .github/CODE_OF_CONDUCT.md]: Great catch, thanks!

**Review by emyarod — APPROVED** [review]
looks good to me!

**Pull request merged**